pkg/internal/blockutil.go

```diff
@@ -0,0 +1,237 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	ExecCommand          = exec.Command
+	FilePathGlob         = filepath.Glob
+	FilePathEvalSymLinks = filepath.EvalSymlinks
+	mountFile            = "/proc/1/mountinfo"
+)
+
+const (
+	// StateSuspended is a possible value of BlockDevice.State
+	StateSuspended = "suspended"
+	// DiskByIDDir is the path for symlinks to the device by id.
+	DiskByIDDir = "/dev/disk/by-id/"
+)
+
+// BlockDevice is the a block device as output by lsblk.
+// All the fields are lsblk columns.
+type BlockDevice struct {
+	Name   string `json:"name"`
+	KName  string `json:"kname"`
+	Type   string `json:"type"`
+	Model  string `json:"model,omitempty"`
+	Vendor string `json:"vendor,omitempty"`
+	State  string `json:"state,omitempty"`
+	FSType string `json:"fsType"`
+	Size   string `json:"size"`
+	// Children   []BlockDevice `json:"children,omitempty"`
+	Rotational string `json:"rota"`
```

Comment:
Having a different names (struct/json) only can drive to confusion. I would suggest to use the same names in both parts... so instead of "rota" use "rotational" , ro/readonly, rm/removable

Reply:
I would take this is up in a follow up PR while working on [this](https://github.com/red-hat-storage/lvm-operator/pull/27#discussion_r773119630) comment